warehouse/models/mart/ntd_validation/fct_ntd_rr20_service_checks.py

```diff
@@ -0,0 +1,223 @@
+import pandas as pd
+import datetime
+import logging
+
+##### TO_DO: see if the missing data check can still work or did we already fill it with zeros
+
+def write_to_log(logfilename):
+    '''
+    Creates a logger object that outputs to a log file, to the filename specified,
+    and also streams to console.
+    '''
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+    formatter = logging.Formatter(f'%(asctime)s:%(levelname)s: %(message)s',
+                                  datefmt='%y-%m-%d %H:%M:%S')
+    file_handler = logging.FileHandler(logfilename)
+    file_handler.setFormatter(formatter)
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(formatter)
+
+    if not logger.hasHandlers():
+        logger.addHandler(file_handler)
+        logger.addHandler(stream_handler)
+
+    return logger
+
+
+def check_rr20_ratios(df, variable, threshold, this_year, last_year, logger):
+    '''Validation checks where a ratio must be within a certain threshold limit
```

Comment:
Is the goal to create columns with these ratio checks, or to treat this as a validation that the rows are valid? (I.e., do you want to have a column with a boolean value for pass/fail, or do you want to just alert if some fail? If the latter, could [leverage dbt tests](https://github.com/dbt-labs/dbt-utils#accepted_range-source).)